---
date: 2015-06-06T12:06:03+08:00
title: 'clusterProfiler: statistical analysis and visualization of functional profiles for genes and gene clusters'
type: index
weight: 1
---



<p><img src="https://raw.githubusercontent.com/Bioconductor/BiocStickers/master/clusterProfiler/clusterProfiler.png" height="200" align="right" /></p>
<p><link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css"></p>
<p><a href="https://bioconductor.org/packages/clusterProfiler"><img src="https://img.shields.io/badge/release%20version-3.8.0-blue.svg?style=flat" alt="releaseVersion" /></a> <a href="https://github.com/guangchuangyu/clusterProfiler"><img src="https://img.shields.io/badge/devel%20version-3.9.0-blue.svg?style=flat" alt="develVersion" /></a> <a href="https://bioconductor.org/packages/stats/bioc/clusterProfiler"><img src="https://img.shields.io/badge/downloads-67607/total-blue.svg?style=flat" alt="total" /></a> <a href="https://bioconductor.org/packages/stats/bioc/clusterProfiler"><img src="https://img.shields.io/badge/downloads-2975/month-blue.svg?style=flat" alt="month" /></a></p>
<p>The <a href="https://bioconductor.org/packages/clusterProfiler/">clusterProfiler</a> package implements methods to analyze and visualize functional profiles of genomic coordinates (supported by ChIPseeker), gene and gene clusters.</p>
<p><a href="https://bioconductor.org/packages/clusterProfiler/">clusterProfiler</a> is released within the <a href="https://bioconductor.org/packages/clusterProfiler">Bioconductor</a> project and the source code is hosted on <a href="https://github.com/GuangchuangYu/clusterProfiler"><i class="fa fa-github fa-lg"></i> GitHub</a>.</p>
<div id="author" class="section level2">
<h2><i class="fa fa-user"></i> Author</h2>
<p>Guangchuang Yu, School of Public Health, The University of Hong Kong.</p>
</div>
<div id="citation" class="section level2">
<h2><i class="fa fa-book"></i> Citation</h2>
<p>Please cite the following article when using <a href="https://bioconductor.org/packages/clusterProfiler/">clusterProfiler</a>:</p>
<p><a href="http://dx.doi.org/10.1089/omi.2011.0118"><img src="https://img.shields.io/badge/doi-10.1089/omi.2011.0118-blue.svg?style=flat" alt="doi" /></a> <a href="https://www.altmetric.com/details/681089"><img src="https://img.shields.io/badge/Altmetric-24-blue.svg?style=flat" alt="Altmetric" /></a> <a href="http://online.liebertpub.com/action/showMostCitedArticles?journalCode=omi"><img src="https://img.shields.io/badge/2nd%20most%20cited%20paper-in%20OMICS-blue.svg?style=flat" /></a> <a href="https://scholar.google.com.hk/scholar?oi=bibs&amp;hl=en&amp;cites=2349076811020942117"><img src="https://img.shields.io/badge/cited%20by-478-blue.svg?style=flat" alt="citation" /></a> <img src="https://img.shields.io/badge/ESI-Highly%20Cited%20Paper-blue.svg?style=flat" /></p>
<p><strong>Yu G</strong>, Wang L, Han Y and He Q<sup>*</sup>. clusterProfiler: an R package for comparing biological themes among gene clusters. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.</p>
</div>
<div id="featured-articles" class="section level2">
<h2><i class="fa fa-pencil"></i> Featured Articles</h2>
<p><img src="https://guangchuangyu.github.io/featured_img/clusterProfiler/elife-02077-fig5-v1.jpg" width="650"></p>
<p><i class="fa fa-hand-o-right"></i> Find out more on <i class="fa fa-pencil"></i> <a href="https://guangchuangyu.github.io/clusterProfiler/featuredArticles/">Featured Articles</a>.</p>
</div>
<div id="installation" class="section level2">
<h2><i class="fa fa-download"></i> Installation</h2>
<p>Install <code>clusterProfiler</code> is easy, follow the guide on the <a href="https://bioconductor.org/packages/clusterProfiler/">Bioconductor page</a>:</p>
<pre class="r"><code>## try http:// if https:// URLs are not supported
source(&quot;https://bioconductor.org/biocLite.R&quot;)
## biocLite(&quot;BiocUpgrade&quot;) ## you may need this
biocLite(&quot;clusterProfiler&quot;)</code></pre>
<p><img src="/index_files/figure-html/unnamed-chunk-2-1.png" width="768" /></p>
</div>
<div id="overview" class="section level2">
<h2><i class="fa fa-cogs"></i> Overview</h2>
<div id="supported-analyses" class="section level4">
<h4><i class="fa fa-angle-double-right"></i> Supported Analyses</h4>
<ul>
<li>Over-Representation Analysis</li>
<li>Gene Set Enrichment Analysis</li>
<li>Biological theme comparison</li>
</ul>
</div>
<div id="supported-ontologiespathways" class="section level4">
<h4><i class="fa fa-angle-double-right"></i> Supported ontologies/pathways</h4>
<ul>
<li>Disease Ontology (via <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li><a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> (via <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li><a href="http://www.disgenet.org/web/DisGeNET/menu/home">DisGeNET</a> (via <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li>Gene Ontology (supports many species with GO annotation query online via <a href="https://bioconductor.org/packages/AnnotationHub/">AnnotationHub</a>)</li>
<li>KEGG Pathway and Module with latest online data (supports more than 4000 species listed in <a href="http://www.genome.jp/kegg/catalog/org_list.html" class="uri">http://www.genome.jp/kegg/catalog/org_list.html</a>)</li>
<li>MeSH enrichment analysis (via <a href="https://www.bioconductor.org/packages/meshes">meshes</a>)</li>
<li>Reactome Pathway (via <a href="https://www.bioconductor.org/packages/ReactomePA">ReactomePA</a>)</li>
<li>DAVID (via <a href="https://www.bioconductor.org/packages/RDAVIDWebService">RDAVIDWebService</a>)</li>
<li><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a>
<ul>
<li>hallmark gene sets</li>
<li>positional gene sets</li>
<li>curated gene sets</li>
<li>motif gene sets</li>
<li>computational gene sets</li>
<li>GO gene sets</li>
<li>oncogenic signatures</li>
<li>immunologic signatures</li>
</ul></li>
<li>Other Annotations
<ul>
<li>from other sources (e.g. <a href="http://www.disgenet.org/web/DisGeNET/menu/home">DisGeNET</a> as <a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">an example</a>)</li>
<li>user’s annotation</li>
<li>customized ontology</li>
<li>and many others</li>
</ul></li>
</ul>
</div>
<div id="visualization" class="section level4">
<h4><i class="fa fa-angle-double-right"></i> Visualization</h4>
<ul>
<li>barplot</li>
<li>cnetplot</li>
<li>dotplot</li>
<li>enrichMap</li>
<li>gseaplot</li>
<li>plotGOgraph (via <a href="https://www.bioconductor.org/packages/topGO">topGO</a> package)</li>
<li>upsetplot</li>
</ul>
</div>
<div id="useful-utilities" class="section level4">
<h4><i class="fa fa-angle-double-right"></i> Useful utilities:</h4>
<ul>
<li>bitr (Biological Id TranslatoR)</li>
<li>compareCluster (biological theme comparison)</li>
<li>dropGO (screen out GO term of specific level or specific term)</li>
<li>go2ont (convert GO ID to Ontology)</li>
<li>go2term (convert GO ID to descriptive term)</li>
<li>gofilter (restrict result at specific GO level)</li>
<li>gsfilter (restrict result by gene set size)</li>
<li>simplify (remove redundant GO terms, supported via <a href="https://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>)</li>
</ul>
<p><i class="fa fa-hand-o-right"></i> Find out details and examples on <i class="fa fa-book"></i> <a href="https://guangchuangyu.github.io/clusterProfiler/documentation/">Documentation</a>.</p>
</div>
</div>
<div id="related-tools" class="section level2">
<h2><i class="fa fa-wrench"></i> Related Tools</h2>
<ul class="fa-ul">
<li>
<i class="fa-li fa fa-angle-double-right"></i><a href="https://guangchuangyu.github.io/DOSE">DOSE</a> for Disease Ontology Semantic and Enrichment analyses
</li>
<li>
<i class="fa-li fa fa-angle-double-right"></i><a href="https://guangchuangyu.github.io/GOSemSim">GOSemSim</a> for GO semantic similarity measurement
</li>
<li>
<i class="fa-li fa fa-angle-double-right"></i><a href="https://guangchuangyu.github.io/meshes">meshes</a> for MeSH Enrichment and Semantic analysis
</li>
<li>
<i class="fa-li fa fa-angle-double-right"></i><a href="https://guangchuangyu.github.io/ReactomePA">ReactomePA</a> for Reactome pathway analysis
</li>
</ul>
<p><i class="fa fa-hand-o-right"></i> Find out more on <a href="https://guangchuangyu.github.io/#projects">projects</a>.</p>
</div>
<div id="projects-that-depend-on-clusterprofiler" class="section level2">
<h2><i class="fa fa-code-fork"></i> Projects that depend on <em>clusterProfiler</em></h2>
<div id="cran-packages" class="section level4">
<h4><i class="fa fa-angle-double-right"></i> CRAN packages</h4>
<ul>
<li><a href="https://cran.r-project.org/package=cRegulome">cRegulome</a>: Obtain and Visualize Regulome-Gene Expression Correlations in Cancer</li>
</ul>
</div>
<div id="bioconductor-packages" class="section level4">
<h4><i class="fa fa-angle-double-right"></i> Bioconductor packages</h4>
<ul>
<li><a href="https://www.bioconductor.org/packages/bioCancer">bioCancer</a>: Interactive Multi-Omics Cancers Data Visualization and Analysis</li>
<li><a href="https://www.bioconductor.org/packages/CEMiTool">CEMiTool</a>: Co-expression Modules identification Tool</li>
<li><a href="https://www.bioconductor.org/packages/DAPAR">DAPAR</a>: Tools for the Differential Analysis of Proteins Abundance with R</li>
<li><a href="https://www.bioconductor.org/packages/debrowser">debrowser</a>: Interactive Differential Expresion Analysis Browser</li>
<li><a href="https://www.bioconductor.org/packages/esATAC">esATAC</a>: An Easy-to-use Systematic pipeline for ATACseq data analysis</li>
<li><a href="https://www.bioconductor.org/packages/GDCRNATools">GDCRNATools</a>: an R/Bioconductor package for integrative analysis of lncRNA, mRNA, and miRNA data in GDC</li>
<li><a href="https://www.bioconductor.org/packages/LINC">LINC</a>: co-expression of lincRNAs and protein-coding genes</li>
<li><a href="https://www.bioconductor.org/packages/MAGeCKFlute">MAGeCKFlute</a>: Integrative analysis pipeline for pooled CRISPR functional genetic screens</li>
<li><a href="https://www.bioconductor.org/packages/miRsponge">miRsponge</a>: Identification and analysis of miRNA sponge interaction networks and modules</li>
<li><a href="https://www.bioconductor.org/packages/TCGAbiolinksGUI">TCGAbiolinksGUI</a>: “TCGAbiolinksGUI: A Graphical User Interface to analyze cancer molecular and clinical data”</li>
</ul>
</div>
<div id="other-applications" class="section level4">
<h4><i class="fa fa-angle-double-right"></i> Other applications</h4>
<ul>
<li><a href="https://github.com/bornea/APOSTL">APOSTL</a>: An Interactive Galaxy Pipeline for Reproducible Analysis of Affinity Proteomics Data</li>
</ul>
</div>
</div>
<div id="feedback" class="section level2">
<h2><i class="fa fa-comment"></i> Feedback</h2>
<ul class="fa-ul">
<li>
<i class="fa-li fa fa-hand-o-right"></i> Please make sure you have followed <a href="https://guangchuangyu.github.io/2016/07/how-to-bug-author/"><strong>the important guide</strong></a> before posting any issue/question
</li>
<li>
<i class="fa-li fa fa-bug"></i> For bugs or feature requests, please post to <i class="fa fa-github-alt"></i> <a href="https://github.com/GuangchuangYu/clusterProfiler/issues">github issue</a>
</li>
<li>
<i class="fa-li fa fa-support"></i> For user questions, please post to <a href="https://support.bioconductor.org/">Bioconductor support site</a> and <a href="https://www.biostars.org/">Biostars</a>. We are following every post tagged with <strong>clusterProfiler</strong>
</li>
<li>
<i class="fa-li fa fa-commenting"></i> Join the group chat on <a href="https://twitter.com/hashtag/clusterProfiler"><i class="fa fa-twitter fa-lg"></i></a> and <a href="http://huati.weibo.com/k/clusterProfiler"><i class="fa fa-weibo fa-lg"></i></a>
</li>
</ul>
</div>
