---
date: 2016-07-10T16:30:13+08:00
title: 'Documentation'
weight: 10
---


<link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css">

## <i class="fa fa-book"></i> Vignettes

+ [DOSE: Disease enrichment analysis](https://bioconductor.org/packages/devel/bioc/vignettes/DOSE/inst/doc/enrichmentAnalysis.html)
+ [DOSE: Disease GSEA](https://bioconductor.org/packages/devel/bioc/vignettes/DOSE/inst/doc/GSEA.html)
+ [Statistical analysis and visualization of functional profiles for genes and gene clusters](https://bioconductor.org/packages/devel/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html)

## <i class="fa fa-wordpress"></i> Blog posts

### <i class="fa fa-angle-double-right"></i> News and updates

+ [clusterProfiler in Bioconductor 2.8](https://guangchuangyu.github.io/2011/03/clusterprofiler-in-bioconductor-2.8)
+ [clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters](https://guangchuangyu.github.io/2012/05/clusterprofiler-an-r-package-for-comparing-biological-themes-among-gene-clusters)
+ [BioC 3.1: NEWS of my BioC packages](https://guangchuangyu.github.io/2015/04/bioc-31-news-of-my-bioc-packages)
+ [BioC 3.2: NEWS of my BioC packages](https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages)
+ [BioC 3.3: NEWS of my BioC packages](https://guangchuangyu.github.io/2016/05/bioc-33-news-of-my-bioc-packages/)
+ [BioC 3.4: NEWS of my BioC packages](https://guangchuangyu.github.io/2016/10/bioc-34-news-of-my-bioc-packages)

### <i class="fa fa-angle-double-right"></i> ID conversion

+ [convert biological ID with KEGG API using clusterProfiler](https://guangchuangyu.github.io/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler)

### <i class="fa fa-angle-double-right"></i> Enrichment analysis

+ [GO analysis using clusterProfiler](https://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler)
+ [use simplify to remove redundancy of enriched GO terms](https://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms)
+ [KEGG Module Enrichment Analysis](https://guangchuangyu.github.io/2016/04/kegg-module-enrichment-analysis)
+ [KEGG enrichment analysis with latest online data using clusterProfiler](https://guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler)
+ [DAVID functional analysis with clusterProfiler](https://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler)
+ [functional enrichment for GTEx paper](https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper)
+ [use clusterProfiler as an universal enrichment analysis tool](https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool)
+ [functional enrichment analysis with NGS data](https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data)
+ [leading edge analysis](https://guangchuangyu.github.io/2016/07/leading-edge-analysis/)
+ [a formula interface for GeneOntology analysis](http://bioinfoblog.it/2015/02/a-formula-interface-for-geneontology-analysis/) <i class="fa fa-arrow-left"></i> bioinfoblog.it
+ [why clusterProfiler fails](https://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails)
+ [Comparison of clusterProfiler and GSEA-P](https://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p)

### <i class="fa fa-angle-double-right"></i> Visualization

+ [dotplot for enrichment result](https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result)
+ [dotplot for GSEA result](http://guangchuangyu.github.io/2016/12/dotplot-for-gsea-result/)
+ [enrichment map](https://guangchuangyu.github.io/2014/08/enrichment-map)
+ [showCategory parameter for visualizing compareCluster output](https://guangchuangyu.github.io/2016/11/showcategory-parameter-for-visualizing-comparecluster-output/)


<i class="fa fa-hand-o-right"></i> Find out more on <https://guangchuangyu.github.io/tags/clusterprofiler>.

## <i class="fa fa-support"></i> Q&A

+ [difference in the number of genes from input and enrichGO results](https://support.bioconductor.org/p/89253/#89272)
+ [Collecting Genes with similar GO term](https://www.biostars.org/p/201064/#201077)
+ [Convert gene to GO ID](https://www.biostars.org/p/215242/#215254)
+ [cnetplot output to Cytoscape](https://www.biostars.org/p/219013/)

## <i class="fa fa-gift"></i> Workflow

+ [An end to end workflow for differential gene expression using Affymetrix microarrays](http://f1000research.com/articles/5-1384/v1)
+ [RNAseq workflow](https://github.com/twbattaglia/RNAseq-workflow)
+ [recount workflow: Accessing over 70,000 human RNA-seq samples with Bioconductor](https://f1000research.com/articles/6-1558/)
+ [TCGA Workflow: Analyze cancer genomics and epigenomics data using Bioconductor packages](http://f1000research.com/articles/5-1542/)

