---
draft: yes
date: 2016-07-10T16:30:13+08:00
title: 'Documentation'
weight: 10
---



<p><link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css"></p>
<div id="vignettes" class="section level2">
<h2><i class="fa fa-book"></i> Vignettes</h2>
<ul>
<li><a href="https://bioconductor.org/packages/devel/bioc/vignettes/DOSE/inst/doc/enrichmentAnalysis.html">DOSE: Disease enrichment analysis</a></li>
<li><a href="https://bioconductor.org/packages/devel/bioc/vignettes/DOSE/inst/doc/GSEA.html">DOSE: Disease GSEA</a></li>
<li><a href="https://bioconductor.org/packages/devel/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html">Statistical analysis and visualization of functional profiles for genes and gene clusters</a></li>
</ul>
</div>
<div id="blog-posts" class="section level2">
<h2><i class="fa fa-wordpress"></i> Blog posts</h2>
<div id="news-and-updates" class="section level3">
<h3><i class="fa fa-angle-double-right"></i> News and updates</h3>
<ul>
<li><a href="https://guangchuangyu.github.io/2011/03/clusterprofiler-in-bioconductor-2.8">clusterProfiler in Bioconductor 2.8</a></li>
<li><a href="https://guangchuangyu.github.io/2012/05/clusterprofiler-an-r-package-for-comparing-biological-themes-among-gene-clusters">clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters</a></li>
<li><a href="https://guangchuangyu.github.io/2015/04/bioc-31-news-of-my-bioc-packages">BioC 3.1: NEWS of my BioC packages</a></li>
<li><a href="https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages">BioC 3.2: NEWS of my BioC packages</a></li>
<li><a href="https://guangchuangyu.github.io/2016/05/bioc-33-news-of-my-bioc-packages/">BioC 3.3: NEWS of my BioC packages</a></li>
<li><a href="https://guangchuangyu.github.io/2016/10/bioc-34-news-of-my-bioc-packages">BioC 3.4: NEWS of my BioC packages</a></li>
</ul>
</div>
<div id="id-conversion" class="section level3">
<h3><i class="fa fa-angle-double-right"></i> ID conversion</h3>
<ul>
<li><a href="https://guangchuangyu.github.io/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler">convert biological ID with KEGG API using clusterProfiler</a></li>
</ul>
</div>
<div id="enrichment-analysis" class="section level3">
<h3><i class="fa fa-angle-double-right"></i> Enrichment analysis</h3>
<ul>
<li><a href="https://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler">GO analysis using clusterProfiler</a></li>
<li><a href="https://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms">use simplify to remove redundancy of enriched GO terms</a></li>
<li><a href="https://guangchuangyu.github.io/2016/04/kegg-module-enrichment-analysis">KEGG Module Enrichment Analysis</a></li>
<li><a href="https://guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler">KEGG enrichment analysis with latest online data using clusterProfiler</a></li>
<li><a href="https://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler">DAVID functional analysis with clusterProfiler</a></li>
<li><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper">functional enrichment for GTEx paper</a></li>
<li><a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool">use clusterProfiler as an universal enrichment analysis tool</a></li>
<li><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data">functional enrichment analysis with NGS data</a></li>
<li><a href="https://guangchuangyu.github.io/2016/07/leading-edge-analysis/">leading edge analysis</a></li>
<li><a href="http://bioinfoblog.it/2015/02/a-formula-interface-for-geneontology-analysis/">a formula interface for GeneOntology analysis</a> <i class="fa fa-arrow-left"></i> bioinfoblog.it</li>
<li><a href="https://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails">why clusterProfiler fails</a></li>
<li><a href="https://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p">Comparison of clusterProfiler and GSEA-P</a></li>
</ul>
</div>
<div id="visualization" class="section level3">
<h3><i class="fa fa-angle-double-right"></i> Visualization</h3>
<ul>
<li><a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result">dotplot for enrichment result</a></li>
<li><a href="http://guangchuangyu.github.io/2016/12/dotplot-for-gsea-result/">dotplot for GSEA result</a></li>
<li><a href="https://guangchuangyu.github.io/2014/08/enrichment-map">enrichment map</a></li>
<li><a href="https://guangchuangyu.github.io/2016/11/showcategory-parameter-for-visualizing-comparecluster-output/">showCategory parameter for visualizing compareCluster output</a></li>
</ul>
<p><i class="fa fa-hand-o-right"></i> Find out more on <a href="https://guangchuangyu.github.io/tags/clusterprofiler" class="uri">https://guangchuangyu.github.io/tags/clusterprofiler</a>.</p>
</div>
</div>
<div id="qa" class="section level2">
<h2><i class="fa fa-support"></i> Q&amp;A</h2>
<ul>
<li><a href="https://support.bioconductor.org/p/89253/#89272">difference in the number of genes from input and enrichGO results</a></li>
<li><a href="https://www.biostars.org/p/201064/#201077">Collecting Genes with similar GO term</a></li>
<li><a href="https://www.biostars.org/p/215242/#215254">Convert gene to GO ID</a></li>
<li><a href="https://www.biostars.org/p/219013/">cnetplot output to Cytoscape</a></li>
</ul>
</div>
<div id="workflow" class="section level2">
<h2><i class="fa fa-gift"></i> Workflow</h2>
<ul>
<li><a href="http://f1000research.com/articles/5-1384/v1">An end to end workflow for differential gene expression using Affymetrix microarrays</a></li>
<li><a href="https://github.com/twbattaglia/RNAseq-workflow">RNAseq workflow</a></li>
<li><a href="https://f1000research.com/articles/6-1558/">recount workflow: Accessing over 70,000 human RNA-seq samples with Bioconductor</a></li>
<li><a href="http://f1000research.com/articles/5-1542/">TCGA Workflow: Analyze cancer genomics and epigenomics data using Bioconductor packages</a></li>
</ul>
</div>
